EasyPost.EasyVCR.Tests/ClientTest.cs

```diff
@@ -6,55 +8,147 @@ namespace EasyPost.EasyVCR.Tests
     [TestClass]
     public class ClientTest
     {
+        private static async Task<List<Post>?> GetPostsRequest(Cassette cassette, Mode mode)
+        {
+            var client = HttpClients.NewHttpClient(cassette, mode);
+
+            var fakeDataService = new FakeDataService(client);
+
+            return await fakeDataService.GetPosts();
+        }
+
+        [TestMethod]
+        public void TestClient()
+        {
+            var client = TestUtils.GetSimpleClient("test_client", Mode.Bypass);
+
+            Assert.IsNotNull(client);
+        }
+
         [TestMethod]
         public void TestFakeDataServiceClient()
         {
-            var cassette = TestUtils.GetCassette("test_fake_data_service_client");
-            var client = HttpClients.NewHttpClient(cassette, Mode.Bypass);
+            var client = TestUtils.GetSimpleClient("test_fake_data_service_client", Mode.Bypass);
 
             var fakeDataService = new FakeDataService(client);
             Assert.IsNotNull(fakeDataService.Client);
         }
 
         [TestMethod]
-        public void TestClient()
+        public async Task TestEraseAndRecord()
         {
-            var cassette = TestUtils.GetCassette("test_client");
-            var client = HttpClients.NewHttpClient(cassette, Mode.Bypass);
+            var cassette = TestUtils.GetCassette("test_erase_and_record");
+            cassette.Erase(); // Erase cassette before recording
+            Assert.IsTrue(cassette.NumInteractions == 0); // Make sure cassette is empty
 
-            Assert.IsNotNull(client);
+            var posts = await GetPostsRequest(cassette, Mode.Record);
+
+            Assert.IsNotNull(posts);
+            Assert.AreEqual(posts.Count, 100);
+            Assert.IsTrue(cassette.NumInteractions > 0); // Make sure cassette is not empty
         }
 
         [TestMethod]
         public async Task TestEraseAndPlayback()
         {
-            var cassette = TestUtils.GetCassette("test_erase_and_playback");
+            var cassette = TestUtils.GetCassette("test_erase_and_record");
             cassette.Erase(); // Erase cassette before recording
             Assert.IsTrue(cassette.NumInteractions == 0); // Make sure cassette is empty
 
-            var client = HttpClients.NewHttpClient(cassette, Mode.Replay);
+            // cassette is empty, so replaying should throw an exception
+            await Assert.ThrowsExceptionAsync<VCRException>(async () => await GetPostsRequest(cassette, Mode.Replay));
+        }
 
-            var fakeDataService = new FakeDataService(client);
+        [TestMethod]
+        public async Task TestAutoMode()
+        {
+            var cassette = TestUtils.GetCassette("test_auto_mode");
+            cassette.Erase(); // Erase cassette before recording
+            Assert.IsTrue(cassette.NumInteractions == 0); // Make sure cassette is empty
+
+            // in replay mode, if cassette is empty, should throw an exception
```

Comment:
Why are we testing two modes in a single function meant for one mode?

Reply:
The replay mode test is just a baseline, to show that auto actually does perform differently than replay (auto attempts replay first (which we see results in failure) before recording)

Reply:
Ok